client/lib/domains/can-domain-owner-add-email.ts

```diff
@@ -0,0 +1,19 @@
+import { getCurrentUserCannotAddEmailReason } from 'calypso/lib/domains';
+import { EMAIL_WARNING_CODE_OTHER_USER_OWNS_EMAIL } from 'calypso/lib/emails/email-provider-constants';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+/**
+ * Determines if domain owner can add email (i.e. domain owner is also the email owner).
+ */
+export function canDomainOwnerAddEmail( domain: ResponseDomain | undefined ) {
```

Comment:
We should add a return type:

```suggestion
export function canDomainOwnerAddEmail( domain: ResponseDomain | undefined ): boolean {
```

Reply:
Is this really necessary? Since Typescript can infer that automatically, I usually only define an explicit return type when there's a special reason for it (e.g. there is some ambiguity, I have defined a custom type or w/e).

Not a big deal either way, but I thought it might be good with some alignment around this.

Reply:
I think that adding a return type just makes the code more explicit and easier to understand. While the name of that function already gives a hint, if the return type is specified I don't have to look at the body of that function to make sure it indeed returns a boolean. And if we always specify return types, then the rule is simple and it means I don't need to worry if I should indeed add one or not when I write a new function.